src/neuroconv/datainterfaces/ophys/femtonics/femtonicsdatainterface.py

```diff
@@ -0,0 +1,245 @@
+        ----------
+        file_path : str or Path
+            Path to the .mesc file.
+        session_index : int, optional
+            Index of the MSession to use (0-based). Default is 0.
+        munit_index : int, optional
+            Index of the MUnit within the specified session (0-based). Default is 0.
+        channel_name : str, optional
+            Name of the channel to extract. If not specified and multiple channels exist,
+            an error will be raised. If only one channel exists, it will be used automatically.
+        verbose : bool, optional
+            Whether to print verbose output. Default is False.
+        """
+        super().__init__(
+            file_path=file_path,
+            session_index=session_index,
+            munit_index=munit_index,
+            channel_name=channel_name,
+            verbose=verbose,
+        )
+
+        self._file_path = file_path
+        self._session_index = session_index
+        self._munit_index = munit_index
+        self._channel_name = channel_name
+
+        # Validate that only one channel is selected
+        channel_names = self.imaging_extractor.get_channel_names()
+        if len(channel_names) != 1:
+            raise ValueError(
+                f"FemtonicsImagingInterface expects a single channel, but found {len(channel_names)}: {channel_names}. "
+                "Please specify 'channel_name' to select one channel."
+            )
+
+        # Hack till roiextractors removes the get_num_channels method in `check_imaging_equal`.
+        # TODO: remove this once roiextractors 0.6.1
+        self.imaging_extractor.get_num_channels = lambda: 1  # Override to ensure only one channel is reported
+
+    def get_metadata(self) -> DeepDict:
+        """
+        Extract metadata specific to Femtonics imaging data.
+
+        Returns
+        -------
+        DeepDict
+            Dictionary containing extracted metadata including device information,
+            optical channels, imaging plane details, and acquisition parameters.
+        """
+        metadata = super().get_metadata()
+
+        femtonics_metadata = self.imaging_extractor._get_metadata()
+
+        # Extract pixel size information for imaging plane
+        pixel_size_info = femtonics_metadata.get("pixel_size_micrometers")
+        if pixel_size_info and "x_size" in pixel_size_info and "y_size" in pixel_size_info:
+            x_size = pixel_size_info["x_size"]
+            y_size = pixel_size_info["y_size"]
+            x_units = pixel_size_info.get("x_units")
+            y_units = pixel_size_info.get("y_units")
+
+            # Only update if both units are the same or if units are missing
+            if x_units == y_units:
+                if "Ophys" in metadata and "ImagingPlane" in metadata["Ophys"]:
+                    for imaging_plane in metadata["Ophys"]["ImagingPlane"]:
+                        imaging_plane["grid_spacing"] = [x_size, y_size]
+                        if x_units:
```

Comment:
So, this checks if the x_units is an empty string. In that case, can you set the unit as "n.a."? I think it is better to be explicit that we don't know what the units is.